src/DynamoCoreWpf/ViewModels/Search/SelectionNavigator.cs

```diff
@@ -0,0 +1,170 @@
+                // We need jump to previous category.
+                if (selectedMemberGroupIndex < 0)
+                {
+                    selectedCategoryIndex += offset;
+
+                    // It's first element of first member group of first category. There is no place to move.
+                    if (selectedCategoryIndex < 0)
+                    {
+                        selectedCategoryIndex = 0;
+                        selectedMemberGroupIndex = 0;
+                        selectedMemberIndex = 0;
+                    }
+                    else
+                    {
+                        // Since we are moving from the bottom to the top, 
+                        // we should select last member and last group.
+                        var cat = root.ElementAt(selectedCategoryIndex);
+                        selectedMemberGroupIndex = cat.MemberGroups.Count() - 1;
+
+                        var group = cat.MemberGroups.ElementAt(selectedMemberGroupIndex);
+
+                        selectedMemberIndex = group.Members.Count() - 1;
+                    }
+                }
+                else
+                {
+                    // Select last member of selected member group.
+                    var group = root.ElementAt(selectedCategoryIndex).MemberGroups.ElementAt(selectedMemberGroupIndex);
+                    selectedMemberIndex = group.Members.Count() - 1;
+                }
+            }
+            else
+            {
+                // Determine the current group size.
+                var members = selectedMemberGroup.Members.Count();
+
+                // It's last member of member group. We can't move further.
+                // We need jump to next member group.
+                if (selectedMemberIndex >= members)
+                {
+                    selectedMemberIndex = 0;
+                    selectedMemberGroupIndex += offset;
+
+                    var memberGroups = root.ElementAt(selectedCategoryIndex).MemberGroups.Count();
+
+                    // It's last member of last member group. We can't move further.
+                    // We need jump to next category.
+                    if (selectedMemberGroupIndex >= memberGroups)
+                    {
+                        selectedMemberGroupIndex = 0;
+                        selectedCategoryIndex += offset;
+
+                        // It's last member of last member group of last category. There is no place to move.
+                        if (selectedCategoryIndex >= root.Count())
+                        {
+                            selectedCategoryIndex = root.Count() - 1;
+                            selectedMemberGroupIndex = root.Last().MemberGroups.Count() - 1;
+                            selectedMemberIndex = root.Last().MemberGroups.Last().Members.Count() - 1;
+                        }
+                    }
+                }
+            }
```

Comment:
Hi Anna, the above code moves forward and then reset back to proper values only when it cannot be moved further. Instead of doing that, a better way would be to try and see if there is any more slot to go or not. If there is, then only update the value, otherwise the values stay. This way it will make the code easier to read.


```csharp
var members = selectedMemberGroup.Members.Count();
if (selectedMemberIndex >= members)
{
    var memberGroups = root.ElementAt(selectedCategoryIndex).MemberGroups.Count();
    if (selectedMemberGroupIndex < memberGroups - 1) // There's still next group.
    {
        selectedMemberIndex = 0;
        selectedMemberGroupIndex ++;
    }
    else if (selectedCategoryIndex < root.Count() - 1) // There's still next category.
    {
        selectedMemberIndex = 0;
        selectedMemberGroupIndex = 0;
        selectedCategoryIndex ++;
    }
}
```

I think the same can be done for moving forward case.